mgmt/MgmtDefs.h

```diff
@@ -61,6 +63,35 @@ typedef enum {
  */
 typedef void *(*MgmtCallback)(void *opaque_cb_data, char *data_raw, int data_len);
 
+//-------------------------------------------------------------------------
+// API conversion functions.
+//-------------------------------------------------------------------------
+/** Conversion functions to and from an aribrary type and Management types.
+ *
+ * A type that wants to support conversion in the TS API should create a static instance of this
+ * class and fill in the appropriate members. The TS API set/get functions can then check for a
+ * @c nullptr to see if the conversion is supported and if so, call a function to do that. The
+ * @c void* argument is a raw pointer to the typed object. For instance, if this is for transaction
+ * overrides the pointer will be to the member in the transaction override configuration structure.
+ * Support for the management types is built in, this is only needed for types that aren't defined
+ * in this header.
+ */
+struct MgmtConverter {
+  // MgmtInt conversions.
+  std::function<MgmtInt(void *)> get_int{nullptr};
+  std::function<void(void *, MgmtInt)> set_int{nullptr};
+
+  // MgmtFloat conversions.
+  std::function<MgmtFloat(void *)> get_float{nullptr};
+  std::function<void(void *, MgmtFloat)> set_float{nullptr};
+
+  // MgmtString conversions.
+  // This is a bit different because it takes std::string_view instead of MgmtString but that's
+  // worth the difference.
+  std::function<std::string_view(void *)> get_string{nullptr};
+  std::function<void(void *, std::string_view)> set_string{nullptr};
+};
```

Comment:
Why not just: https://godbolt.org/g/qtR5w7 ?

Do you need to wear a drool bucket every time you think of std::function?

Reply:
Your code doesn't compile if any of the lambdas do a capture.

Reply:
I think captures are only useful for local variables, and I think the instances of this class would never be a local variable, thus no need / point in captures.

Reply:
It's not performant code so I'd rather have the flexibility. There are already layers of virtual dispatches to get to this point, it's not going to make a detectable difference. You should get over your allergic reaction to `std::function`.

I'm also not confident local variables won't be needed - I can easily see wanting to embed references to other objects needed to do the conversions, such as name tables and such.

Reply:
Captures are only necessary to embed references to local variables I think.

Reply:
`std::function` is cleaner and simpler and therefore should be used by default, unless there is a good reason to write more complicated code, such as for tight performance constraints.

Reply:
I don't see how using std::function is simpler than just a function pointer when that will do.

Reply:
As noted, I am not confident a simple function pointer will do. I ended up doing that inside some of the buffer writer formatting and now it's a nasty problem preventing me from doing things I want to do. I think micro-optimizing, especially in cases like this where performance isn't critical, is usually a bad choice.